ansible/roles/telegraf/handlers/main.yml

```diff
@@ -0,0 +1,9 @@
+---
+# Copyright (C) 2021 Dr. Serge Victor <https://dr.sergevictor.eu/>
+# Copyright (C) 2021 DebOps <https://debops.org/>
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+- name: Restart Telegraf
+  service:
+    name: "telegraf"
+    state: "restarted"
```

Comment:
You should put the handler in the `global_handlers` role, that way other roles can integrate with the Telegraf service and reload/restart it if necessary. Also, usually handlers are called with lowercase service name, so "Restart telegraf". Remember to update the `notify` keywords as well.

Does Telegraf provide a way to check configuration for errors? It might be a good idea to check it before restarting to avoid disrupting the service in case of wrong configuration provided by the user.